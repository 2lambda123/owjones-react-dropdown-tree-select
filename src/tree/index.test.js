import React from 'react'
import renderer from 'react-test-renderer'

import Tree from './index'
import TreeManager from '../tree-manager'

describe('<Tree/>', () => {
  it('renders tree nodes when search mode is on', () => {
    const treeData = [
      {
        label: 'item1',
        value: 'value1',
        children: [
          {
            label: 'item1-1',
            value: 'value1-1',
            children: [{ label: 'item1-1-1', value: 'value1-1-1' }, { label: 'item1-1-2', value: 'value1-1-2' }]
          },
          { label: 'item1-2', value: 'value1-2' }
        ]
      },
      {
        label: 'item2',
        value: 'value2',
        children: [
          {
            label: 'item2-1',
            value: 'value2-1',
            children: [
              { label: 'item2-1-1', value: 'value2-1-1' },
              { label: 'item2-1-2', value: 'value2-1-2' },
              { label: 'item2-1-3', value: 'value2-1-3', children: [{ label: 'item2-1-3-1', value: 'value2-1-3-1' }] }
            ]
          },
          { label: 'item2-2', value: 'value2-2' }
        ]
      }
    ]
    const treeManager = new TreeManager(treeData)
    const tree = renderer
      .create(<Tree data={treeManager.tree} searchModeOn />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders only expanded tree nodes when search mode is off', () => {
    const treeData = [
      {
        label: 'item1',
        value: 'value1',
        expanded: true,
        className: 'should-be-rendered',
        children: [
          {
            label: 'item1-1',
            value: 'value1-1',
            className: 'should-be-rendered',
            children: [
              { label: 'item1-1-1', value: 'value1-1-1', className: 'should-not-be-rendered' },
              { label: 'item1-1-2', value: 'value1-1-2', className: 'should-not-be-rendered' }
            ]
          },
          { label: 'item1-2', value: 'value1-2', className: 'should-be-rendered' }
        ]
      },
      {
        label: 'item2',
        value: 'value2',
        className: 'should-be-rendered',
        children: [
          {
            label: 'item2-1',
            value: 'value2-1',
            className: 'should-not-be-rendered',
            children: [
              { label: 'item2-1-1', value: 'value2-1-1', className: 'should-not-be-rendered' },
              { label: 'item2-1-2', value: 'value2-1-2', className: 'should-not-be-rendered' },
              {
                label: 'item2-1-3',
                value: 'value2-1-3',
                className: 'should-not-be-rendered',
                children: [{ label: 'item2-1-3-1', value: 'value2-1-3-1', className: 'should-not-be-rendered' }]
              }
            ]
          },
          {
            label: 'item2-2',
            value: 'value2-2',
            className: 'should-not-be-rendered'
          }
        ]
      }
    ]

    const treeManager = new TreeManager(treeData)
    const tree = renderer
      .create(<Tree data={treeManager.tree} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
