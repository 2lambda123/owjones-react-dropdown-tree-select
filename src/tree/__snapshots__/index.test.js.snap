// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tree/> renders only expanded tree nodes when search mode is off 1`] = `
<ul
  className="root "
>
  <li
    className="node tree should-be-rendered"
    style={
      Object {
        "paddingLeft": "0px",
      }
    }
  >
    <i
      className="toggle expanded"
      onClick={[Function]}
    />
    <label
      title="item1"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="0"
        onChange={[Function]}
        type="checkbox"
        value="value1"
      />
      <span
        className="node-label"
      >
        item1
      </span>
    </label>
  </li>
  <li
    className="node tree should-be-rendered"
    style={
      Object {
        "paddingLeft": "20px",
      }
    }
  >
    <i
      className="toggle collapsed"
      onClick={[Function]}
    />
    <label
      title="item1-1"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="0-0"
        onChange={[Function]}
        type="checkbox"
        value="value1-1"
      />
      <span
        className="node-label"
      >
        item1-1
      </span>
    </label>
  </li>
  <li
    className="node leaf should-be-rendered"
    style={
      Object {
        "paddingLeft": "20px",
      }
    }
  >
    <i
      className="toggle"
      onClick={[Function]}
    />
    <label
      title="item1-2"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="0-1"
        onChange={[Function]}
        type="checkbox"
        value="value1-2"
      />
      <span
        className="node-label"
      >
        item1-2
      </span>
    </label>
  </li>
  <li
    className="node tree should-be-rendered"
    style={
      Object {
        "paddingLeft": "0px",
      }
    }
  >
    <i
      className="toggle collapsed"
      onClick={[Function]}
    />
    <label
      title="item2"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="1"
        onChange={[Function]}
        type="checkbox"
        value="value2"
      />
      <span
        className="node-label"
      >
        item2
      </span>
    </label>
  </li>
</ul>
`;

exports[`<Tree/> renders tree nodes when search mode is on 1`] = `
<ul
  className="root searchModeOn"
>
  <li
    className="node tree"
    style={Object {}}
  >
    <i
      className="toggle collapsed"
      onClick={[Function]}
    />
    <label
      title="item1"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="0"
        onChange={[Function]}
        type="checkbox"
        value="value1"
      />
      <span
        className="node-label"
      >
        item1
      </span>
    </label>
  </li>
  <li
    className="node tree"
    style={Object {}}
  >
    <i
      className="toggle collapsed"
      onClick={[Function]}
    />
    <label
      title="item1-1"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="0-0"
        onChange={[Function]}
        type="checkbox"
        value="value1-1"
      />
      <span
        className="node-label"
      >
        item1-1
      </span>
    </label>
  </li>
  <li
    className="node leaf"
    style={Object {}}
  >
    <i
      className="toggle"
      onClick={[Function]}
    />
    <label
      title="item1-1-1"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="0-0-0"
        onChange={[Function]}
        type="checkbox"
        value="value1-1-1"
      />
      <span
        className="node-label"
      >
        item1-1-1
      </span>
    </label>
  </li>
  <li
    className="node leaf"
    style={Object {}}
  >
    <i
      className="toggle"
      onClick={[Function]}
    />
    <label
      title="item1-1-2"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="0-0-1"
        onChange={[Function]}
        type="checkbox"
        value="value1-1-2"
      />
      <span
        className="node-label"
      >
        item1-1-2
      </span>
    </label>
  </li>
  <li
    className="node leaf"
    style={Object {}}
  >
    <i
      className="toggle"
      onClick={[Function]}
    />
    <label
      title="item1-2"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="0-1"
        onChange={[Function]}
        type="checkbox"
        value="value1-2"
      />
      <span
        className="node-label"
      >
        item1-2
      </span>
    </label>
  </li>
  <li
    className="node tree"
    style={Object {}}
  >
    <i
      className="toggle collapsed"
      onClick={[Function]}
    />
    <label
      title="item2"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="1"
        onChange={[Function]}
        type="checkbox"
        value="value2"
      />
      <span
        className="node-label"
      >
        item2
      </span>
    </label>
  </li>
  <li
    className="node tree"
    style={Object {}}
  >
    <i
      className="toggle collapsed"
      onClick={[Function]}
    />
    <label
      title="item2-1"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="1-0"
        onChange={[Function]}
        type="checkbox"
        value="value2-1"
      />
      <span
        className="node-label"
      >
        item2-1
      </span>
    </label>
  </li>
  <li
    className="node leaf"
    style={Object {}}
  >
    <i
      className="toggle"
      onClick={[Function]}
    />
    <label
      title="item2-1-1"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="1-0-0"
        onChange={[Function]}
        type="checkbox"
        value="value2-1-1"
      />
      <span
        className="node-label"
      >
        item2-1-1
      </span>
    </label>
  </li>
  <li
    className="node leaf"
    style={Object {}}
  >
    <i
      className="toggle"
      onClick={[Function]}
    />
    <label
      title="item2-1-2"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="1-0-1"
        onChange={[Function]}
        type="checkbox"
        value="value2-1-2"
      />
      <span
        className="node-label"
      >
        item2-1-2
      </span>
    </label>
  </li>
  <li
    className="node tree"
    style={Object {}}
  >
    <i
      className="toggle collapsed"
      onClick={[Function]}
    />
    <label
      title="item2-1-3"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="1-0-2"
        onChange={[Function]}
        type="checkbox"
        value="value2-1-3"
      />
      <span
        className="node-label"
      >
        item2-1-3
      </span>
    </label>
  </li>
  <li
    className="node leaf"
    style={Object {}}
  >
    <i
      className="toggle"
      onClick={[Function]}
    />
    <label
      title="item2-1-3-1"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="1-0-2-0"
        onChange={[Function]}
        type="checkbox"
        value="value2-1-3-1"
      />
      <span
        className="node-label"
      >
        item2-1-3-1
      </span>
    </label>
  </li>
  <li
    className="node leaf"
    style={Object {}}
  >
    <i
      className="toggle"
      onClick={[Function]}
    />
    <label
      title="item2-2"
    >
      <input
        checked={undefined}
        className="checkbox-item"
        disabled={undefined}
        name="1-1"
        onChange={[Function]}
        type="checkbox"
        value="value2-2"
      />
      <span
        className="node-label"
      >
        item2-2
      </span>
    </label>
  </li>
</ul>
`;
